src/tallies/filter_distribcell.cpp

```diff
@@ -0,0 +1,82 @@
+#include "openmc/tallies/filter_distribcell.h"
+
+#include "openmc/cell.h"
+#include "openmc/error.h"
+#include "openmc/geometry_aux.h" // For distribcell_path
+#include "openmc/lattice.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+void
+DistribcellFilter::from_xml(pugi::xml_node node)
+{
+  auto cells = get_node_array<int32_t>(node, "bins");
+  if (cells.size() != 1) {
+    fatal_error("Only one cell can be specified per distribcell filter.");
+  }
+
+  // Find index in global cells vector corresponding to cell ID
+  auto search = model::cell_map.find(cells[0]);
+  if (search == model::cell_map.end()) {
+    std::stringstream err_msg;
+    err_msg << "Could not find cell " << cell_
+            << " specified on tally filter.";
+    throw std::runtime_error{err_msg.str()};
+  }
+
+  this->set_cell(search->second);
+}
+
+void
+DistribcellFilter::set_cell(int32_t cell)
+{
+  Expects(cell >= 0);
```

Comment:
Why is expects only used here and not in other error handling? We may want be more consistent.